File: ROTn.c
#include <stdio.h>

int main() {
    int rot;
    int length;
    printf("ROT? > ");
    scanf("%d", &rot);
    printf("length > ");
    scanf("%d", &length);
    char str[length];
    printf("string & $ > ");
    scanf("%[^$}", &str);
    for (int i = 0; i < length; i++) {
        if ((str[i] >= 65 && str[i] <= 90 - rot) || (str[i] >= 97 && str[i] <= 122 - rot)) str[i] += rot;
        else if ((str[i] > 90 - rot && str[i] <= 90) || (str[i] > 122 - rot && str[i] <= 122)) str[i] -= rot;
    }
    printf("%s", str);
    return 0;
}

File: ROTn.py
import sys

rot = int(input('ROT? > '))
string = input('String > ')
chr_list = list(string)
for character in chr_list:
    if 65 <= ord(character) <= 90 - rot or 97 <= ord(character) <= 122 - rot:
        sys.stdout.write(chr(ord(character) + 13))
    elif 90 - rot < ord(character) <= 90 or 122 - rot < ord(character) <= 122:
        sys.stdout.write(chr(ord(character) - 13))
    else:
        sys.stdout.write(character)
